backend/src/test/java/com/staccato/member/domain/NicknameTest.java

```diff
@@ -0,0 +1,36 @@
+package com.staccato.member.domain;
+
+import static org.assertj.core.api.Assertions.assertThatNoException;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import com.staccato.exception.StaccatoException;
+
+class NicknameTest {
+    @DisplayName("유효한닉네임을 생성한다.")
+    @Test
+    void CreateNickname() {
+        assertThatNoException().isThrownBy(() -> new Nickname("가ㄱㅏㅣㅎ.AZaz_"));
+    }
+
+    @DisplayName("닉네임의 길이가 잘못되었을 경우 예외를 발생시킨다.")
+    @ParameterizedTest
+    @ValueSource(ints = {0, 21})
+    void cannotCreateNicknameByInvalidLength(int length) {
+        assertThatThrownBy(() -> new Nickname("가".repeat(length)))
+                .isInstanceOf(StaccatoException.class)
+                .hasMessage("20자 이내의 닉네임으로 설정해주세요.");
+    }
+
+    @DisplayName("닉네임의 형식이 잘못되었을 경우 예외를 발생시킨다.")
+    @Test
+    void cannotCreateNicknameByInvalidFormat() {
+        assertThatThrownBy(() -> new Nickname("//"))
+                .isInstanceOf(StaccatoException.class)
+                .hasMessage("올바르지 않은 닉네임 형식입니다.");
+    }
```

Comment:
찾아보니 닉네임을 입력받을 때, 주의해야 할 보안 사항으로 아래와 같은 문제점들이 있네요. 왜 닉네임 정규표현식을 이런 식으로 정했는지에 대해 이유도 다같이 알고 넘어가면 좋을 것 같아서 아래에 간단히 조사한 내용만 기록해놓겠습니다. 

좀 더 깊은 공부나 그 이후 정규표현식 변경은 나중에 다같이 해봐요!

---

### SQL Injection

사용자가 로그인할 때 입력한 사용자명과 비밀번호를 데이터베이스에서 검증하는 SQL 쿼리가 있다고 가정합니다.

```sql
SELECT * FROM users WHERE username = 'admin' AND password = 'password';
```

공격자가 사용자명에 admin' --을 입력하면, SQL 쿼리는 다음과 같이 변경됩니다.

```sql
SELECT * FROM users WHERE username = 'admin' -- ' AND password = 'password';
```

-- 이후의 모든 내용은 주석 처리되어 무시됩니다. 따라서 쿼리는 항상 참이 되어 데이터베이스의 모든 사용자 정보를 반환할 수 있습니다.

---

### XSS(Cross-Site Scripting)

사용자가 다음과 같은 악성 스크립트를 댓글로 입력할 수 있습니다.

```html
<script>alert('XSS Attack');</script>
```

이 댓글이 그대로 웹 페이지에 포함되면, 페이지를 방문한 모든 사용자의 브라우저에서 해당 스크립트가 실행됩니다.

Reply:
👍👍
물론! 저희가 아직 할 작업은 아니겠지만(ㅎㅎ), 추가로 읽어볼만한 자료 공유합니다.
[SQL Injection and How to Prevent It?](https://www.baeldung.com/sql-injection)
[Prevent Cross-Site Scripting (XSS) in a Spring Application](https://www.baeldung.com/spring-prevent-xss)